### PR TITLE
ui/breadcrumbs: allow lualine winbar component to be disabled

### PR DESCRIPTION
**Pull request by NotAShelf**
Adds `vim.ui.breadcrumbs.lualine.winbar.enable` to control winbar component behaviour, and moves `vim.ui.breadcrumbs.alwaysRender` (whether to always render the winbar compontent) to `vim.ui.breadcrumbs.lualine.winbar.alwaysRender`

### CODE REVIEW

File: modules/plugins/statusline/lualine/config.nix [hunk 1]
```diff
@@ -11,7 +11,7 @@
   inherit (lib.generators) mkLuaInline;
 
   cfg = config.vim.statusline.lualine;
-  breadcrumbsCfg = config.vim.ui.breadcrumbs;
+  bcfg = config.vim.ui.breadcrumbs;
```
Comment (diniamo) [inline modules/plugins/statusline/lualine/config.nix:14]: Highly meh naming

File: modules/plugins/statusline/lualine/config.nix [hunk 2]
```diff
@@ -20,13 +20,13 @@ in {
         extensions = ["nvim-tree"];
       };
     })
-    (mkIf (breadcrumbsCfg.enable && breadcrumbsCfg.source == "nvim-navic") {
+    (mkIf (bcfg.enable && bcfg.source == "nvim-navic" && bcfg.lualine.winbar.enable) {
```
Comment (diniamo) [inline modules/plugins/statusline/lualine/config.nix:23]: Maybe group the 2  enable options together, makes a little more sense, eg.: `bcfg.enable && bcfg.lualine.winbar.enable && bcfg.source == "nvim-navic`

File: modules/plugins/ui/breadcrumbs/breadcrumbs.nix [hunk 3]
```diff
@@ -43,17 +45,33 @@ in {
...
+        ''
+        // {default = true;}; # for retaining previous behaviour
+
+      alwaysRender = mkOption {
+        type = bool;
+        default = true;
+        example = false;
+        description = ''
+          Whether to always display the breadcrumbs component
+          on winbar (always renders winbar.)
```
Comment (diniamo) [inline modules/plugins/ui/breadcrumbs/breadcrumbs.nix:68]: I don't know what the addition of the parentheses comment is actually adding here, maybe change it to `(even in empty buffers)

File: modules/plugins/ui/breadcrumbs/breadcrumbs.nix [hunk 4]
```diff
@@ -43,17 +45,33 @@ in {
...
+
+          ::: {.note}
+          This is **set to `true` by default**, which means nvim-navic
+          will occupy `winbar.lualine_c` for the breadcrumbs feature
+          unless this option is set to `false`.
+          :::
+        ''
+        // {default = true;}; # for retaining previous behaviour
+
+      alwaysRender = mkOption {
```
Comment (diniamo) [inline modules/plugins/ui/breadcrumbs/breadcrumbs.nix:62]: Could use mkEnableOption here too

**Review by diniamo — CHANGES_REQUESTED** [review]

File: modules/plugins/statusline/lualine/config.nix (hunk 1 above)
Comment (NotAShelf) [reply modules/plugins/statusline/lualine/config.nix:14]: highly doesn't matter :trollface: 

what should it be? breadcrumbsCfg is way too long

File: modules/plugins/ui/breadcrumbs/breadcrumbs.nix (hunk 4 above)
Comment (NotAShelf) [reply modules/plugins/ui/breadcrumbs/breadcrumbs.nix:62]: Could, I just didn't want to deal with "Whether to enable..." suffix. It's hard to make sense with.

File: modules/plugins/ui/breadcrumbs/breadcrumbs.nix (hunk 3 above)
Comment (NotAShelf) [reply modules/plugins/ui/breadcrumbs/breadcrumbs.nix]: Updated description to properly describe the result

File: modules/plugins/statusline/lualine/config.nix (hunk 1 above)
Comment (diniamo) [reply modules/plugins/statusline/lualine/config.nix:14]: Keep bcfg I guess, but at least capitalize the C?

Comment (NotAShelf) [reply modules/plugins/statusline/lualine/config.nix]: done

**Review by diniamo — APPROVED** [review]

**Pull request merged**